src/server/internalServer.cpp

```diff
@@ -109,6 +109,66 @@ unsigned int getCacheLength(const char* name, unsigned int defaultVal) {
 }
 } // unnamed namespace
 
+SearchInfo::SearchInfo(const std::string& pattern)
+  : pattern(pattern),
+    has_geo_query(false),
+    geo_query(0,0,0)
+{}
+
+SearchInfo::SearchInfo(const std::string& pattern, GeoQuery geo_query)
+  : pattern(pattern),
+    has_geo_query(true),
+    geo_query(geo_query)
+{}
+
+SearchInfo::SearchInfo(const RequestContext& request)
+  : pattern(""),
+    has_geo_query(false),
+    geo_query(0, 0, 0)
+{
+  try {
+    pattern = request.get_argument("pattern");
+  } catch (const std::out_of_range&) {}
+
+  /* Retrive geo search */
+  try {
+    auto latitude = request.get_argument<float>("latitude");
+    auto longitude = request.get_argument<float>("longitude");
+    auto distance = request.get_argument<float>("distance");
+    geo_query = GeoQuery(latitude, longitude, distance);
+    has_geo_query = true;
+  } catch(const std::out_of_range&) {}
+    catch(const std::invalid_argument&) {}
+
+  if (!has_geo_query && pattern.empty()) {
+    throw std::invalid_argument("No query provided.");
+  }
+
+  try {
+    bookName = request.get_argument("content");
+  } catch (const std::out_of_range&) {}
+}
+
+zim::Query SearchInfo::getZimQuery(bool verbose) const {
+  zim::Query query;
+  if(pattern.empty()) {
+    // Execute geo-search
+    if (verbose) {
+      cout << "Performing geo query `" << geo_query.distance << "&(" << geo_query.latitude << ";" << geo_query.longitude << ")'" << endl;
+    }
+    query.setQuery("");
+    query.setGeorange(geo_query.latitude, geo_query.longitude, geo_query.distance);
+  } else {
+    // Execute Ft search
+    if (verbose) {
+        cout << "Performing query `" << pattern << "'" << endl;
+    }
+    query.setQuery(pattern);
+  }
+  return query;
+}
```

Comment:
Such an implementation of `SearchInfo::getZimQuery()` makes the `SearchInfo::SearchInfo(const std::string& pattern, GeoQuery geo_query)` constructor meaningless

Reply:
This is what was done previously. But yes, it is better to always use the pattern if provided.
Changed in a new commit.